rust/kernel/file_operations.rs

```diff
@@ -21,14 +21,26 @@ pub struct File {
 }
 
 impl File {
+    /// Constructors a new [`struct file`] wrapper.
+    ///
+    /// # Safety
+    ///
+    /// The pointer `ptr` must be non-null and valid for the lifetime of the object.
     unsafe fn from_ptr(ptr: *const bindings::file) -> File {
         File { ptr }
     }
```

Comment:
Since this looks like a type invariant, let's give a try to the convention I suggested the other day:

```suggestion
    unsafe fn from_ptr(ptr: *const bindings::file) -> File {
        // INVARIANTS: the safety contract ensures the type invariant.
        File { ptr }
    }
```

Then in the `File` docs:

    /// # Invariants
    ///
    /// The pointer `ptr` is non-null and valid.

and then in the `SAFETY` comments below, we can say:

    // SAFETY: `File::ptr` is guaranteed to be valid by the type invariants.

Reply:
Sure.